compiler/noirc_evaluator/src/ssa/opt/flatten_cfg/branch_analysis.rs

```diff
@@ -72,54 +109,169 @@ impl<'cfg> Context<'cfg> {
     ) -> BasicBlockId {
         let left = successors.next().unwrap();
         let right = successors.next().unwrap();
+        let mut visited = HashSet::new();
+
+        // Kick off the stack from the starting branch. It doesn't have a parent.
+        self.stack.push((start, left, right));
```

Comment:
Is the stack cleared after this method returns? I'm seeing that this method is called in a `loop` so it might be called again and I'm not sure it will start with a fresh state.

Reply:
Yes, the `while` loop goes until it’s empty. At some point I started with an assertion that it’s empty, but I thought since there is no early return it’s easy to see that this is the case.

Reply:
Done, also included the illustration from below.